README.md

```diff
@@ -1,61 +1,419 @@
+```swift
+class Unreliable: Context {
+    func execute(_ call: @escaping () -> ()) {
+        /// could call more than once
+    }
+}
 
-Sourcery will automatically generate tests for all types conforming to the protocols representing the algebraic data structures considered, and some annotations associated with a type (in the form `// sourcery: annotation`) will allow some fine tuning or `Arbitrary` generation:
+class Distributor {
+    let unreliableContext = Unreliable()
+    
+    func distribute<T>(process: Process<T>, to collector: Collector<T>) where T: BoundedSemilattice {
+        ProcessingUnit.init(context: unreliableContext).execute(process, onComplete: collector.append)
+    }
+}
+```
 
-- `ignore = "value"`: Sourcery will not generate tests related to that protocol for that type; "value" is the protocol's name (like "Semigroup" or "CommuntativeMonoid");
-- `fixedTypesForPropertyBasedTests = "value"`: for generic types it defines the concrete type to be used in tests; if more than one generic type is present, all types must be separated by a comma;
-- `requiredContextForPropertyBasedTests = "value"`: use `LawInContext` instead of `Law`; "value" is the context type; this is required if the type wraps a function;
-- `arbitrary`: Sourcery will generate the `Arbitrary` definitions for that type; if it's a non-generic type Sourcery will add a `extension Type: Arbitrary`; if the type is generic, Sourcery will define a new `struct ArbitraryType<T: Arbitrary>: Arbitrary`;
-- `arbitraryFunction`: same as `arbitrary` but for function wrappers (like `FunctionS`);
-- `arbitraryGenericParameterProtocols = "value"`: if the type is generic, Sourcery will add these constraints for the generic parameter `T` in `ArbitraryType<T>` in the form of `T: value`
-- `additionalGenericParameterSubtypeRequirement = "value"`: if the type is generic, and a subtype of the generic paramter has some additional constraints, Sourcery will add them to `ArbitraryType<T>` in the form of `T.value` (example: `T.Additive: Equatable` if "value" is `Additive: Equatable`);
-- `additionalGenericParameterSubtypeRequirements = "value"`: same as before, but there's more than one requirement; in this case, more than one `additionalGenericParameterSubtypeRequirements` annotation should be added; if there's only one requirement, `additionalGenericParameterSubtypeRequirement` should be used.
+By clearly defining the composition semantics of `T` we can make assumptions that allow us to be more efficient and less constrained. But making `T` conform to `BoundedSemilattice`, `CommutativeMonoid` or even `Monoid` will require that the composition operation on `T` respects some laws like *commutativity*: `Abstract` provides functions, defined in the `Law` namespace, that will allow one to test a type against these laws, thus proving that the type has the required semantics.
```

Comment:
All of these, from here on up, belongs in a tutorial playground.

Reply:
Not sure about this. I can add a tutorial playground in which I rewrite these examples, but the readme as it is reads like an article in which the concepts are explored, and could work as a reference for people that are just interested in the basic theory.

Also, I think that the library is still not complete enough for a proper `Tutorial.playground`: the testing part is fine, but there's not enough utility types/functions to make this library a desirable drop-in (yet).

I came up with those examples for the README because simpler examples with numbers and/or strings didn't seem relevant enough to me: do you want me to remove them from the README?